input/postgres/statements.go

```diff
@@ -234,6 +235,20 @@ func GetStatements(server *state.Server, logger *util.Logger, db *sql.DB, global
 		if showtext {
 			statementTexts[key] = receivedQuery.String
 		}
+		if queryID.Valid && showtext {
+			if server.PrevState.QueryIdentities == nil {
+				server.PrevState.QueryIdentities = make(state.QueryIdentityMap)
+			}
+			if identity, ok := server.PrevState.QueryIdentities[queryID.Int64]; ok {
+				identity.LastSeen = time.Now()
+			} else {
+				server.PrevState.QueryIdentities[queryID.Int64] = state.QueryIdentity{
+					QueryID:     queryID.Int64,
+					Fingerprint: util.FingerprintQuery(receivedQuery.String, server.Config.FilterQueryText, -1),
+					LastSeen:    time.Now(),
+				}
```

Comment:
We generally don't mutate `server.PrevState` in the input functions - is there a way to write this, that avoids this?

(maybe `QueryIdentities` shouldn't be in state after all, but rather be a separate variable on the `Server` struct? That way we could also have a separate mutex for it, making it easier to have concurrent activity snapshots access its information)

Reply:
Would we write the cache to a separate file or skip persisting it? I could see arguments for both: saving it would make missing fingerprints much less likely if the collector restarts regularly (OOM), though if there was some undiscovered bug it at least wouldn't stick around across collector version updates.

Reply:
Good question - I think we could do it either way. Probably doesn't hurt to persist it.

If we wanted to persist it, we could simply add it to the `StateOnDisk` struct and update `WriteStateFile` and `ReadStateFile` accordingly - i.e. still keep it as a separate value on the `Server` struct itself (not have it inside state, per the above mentioned mutex issue).